src/transformation/visitors/class/index.ts

```diff
@@ -197,7 +193,14 @@ export function transformClassDeclaration(
 
     if (!isExtension && !isMetaExtension) {
         result.push(
-            ...createClassSetup(context, classDeclaration, className, localClassName, classNameText, extendedType)
+            ...createClassSetup(
+                context,
+                classDeclaration,
+                className,
+                localClassName,
+                getReflectionClassName(classDeclaration),
```

Comment:
Should we move `getReflectionClassName` implementation to setup instead of passing it as a parameter? Unrelated, buy maybe `extendedType` too?